pkg/api/server/v1alpha2/records_test.go

```diff
@@ -245,6 +246,258 @@ func TestGetRecord(t *testing.T) {
 	}
 }
+func TestListRecordsV2(t *testing.T) {
+	// Create a temporary database
+	srv, err := New(&config.Config{DB_ENABLE_AUTO_MIGRATION: true}, logger.Get("info"), test.NewDB(t))
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+	ctx := context.Background()
+
+	result, err := srv.CreateResult(ctx, &pb.CreateResultRequest{
+		Parent: "foo",
+		Result: &pb.Result{
+			Name: "foo/results/bar",
+		},
+	})
+	if err != nil {
+		t.Fatalf("CreateResult: %v", err)
+	}
+
+	records := make([]*pb.Record, 0, 6)
+	// Create 3 TaskRun records
+	for i := 0; i < 3; i++ {
+		fakeClock.Advance(time.Second)
+		r, err := srv.CreateRecord(ctx, &pb.CreateRecordRequest{
+			Parent: result.GetName(),
+			Record: &pb.Record{
+				Name: fmt.Sprintf("%s/records/%d", result.GetName(), i),
+				Data: &pb.Any{
+					Type: "TaskRun",
+					Value: jsonutil.AnyBytes(t, &v1beta1.TaskRun{ObjectMeta: v1.ObjectMeta{
+						Name: fmt.Sprintf("%d", i),
+					}}),
+				},
+			},
+		})
+		if err != nil {
+			t.Fatalf("could not create result: %v", err)
+		}
+		t.Logf("Created record: %+v", r)
+		records = append(records, r)
+	}
+
+	// Create 3 PipelineRun records
+	for i := 3; i < 6; i++ {
+		fakeClock.Advance(time.Second)
+		r, err := srv.CreateRecord(ctx, &pb.CreateRecordRequest{
+			Parent: result.GetName(),
+			Record: &pb.Record{
+				Name: fmt.Sprintf("%s/records/%d", result.GetName(), i),
+				Data: &pb.Any{
+					Type: "PipelineRun",
+					Value: jsonutil.AnyBytes(t, &v1beta1.PipelineRun{ObjectMeta: v1.ObjectMeta{
+						Name: fmt.Sprintf("%d", i),
+					}}),
+				},
+			},
+		})
+		if err != nil {
+			t.Fatalf("could not create result: %v", err)
+		}
+		t.Logf("Created record: %+v", r)
+		records = append(records, r)
+	}
+
+	reversedRecords := make([]*pb.Record, len(records))
+	for i := len(reversedRecords); i > 0; i-- {
+		reversedRecords[len(records)-i] = records[i-1]
+	}
+
+	tt := []struct {
+		name   string
+		req    *pb.ListRecordsRequest
+		want   *pb.ListRecordsResponse
+		status codes.Code
+	}{
+		{
+			name: "all",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			name: "list all records without knowing the result name",
+			req: &pb.ListRecordsRequest{
+				Parent: "foo/results/-",
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			name: "list all records without knowing the parent and the result name",
+			req: &pb.ListRecordsRequest{
+				Parent: "-/results/-",
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			// TODO: We should return NOT_FOUND in the future.
+			name: "missing parent",
+			req: &pb.ListRecordsRequest{
+				Parent: "foo/results/baz",
+			},
+			status: codes.NotFound,
+		},
+		{
+			name: "filter by record property",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+				// Filter: `name == "foo/results/bar/records/0"`,
+				Filter: `name == "0"`,
```

Comment:
The `name` we used to expose via CEL is not the same as the database. Should we worry about this?